ci/infra/testrunner/platforms/platform.py

```diff
@@ -1,28 +1,100 @@
+import os
 import sys
 
+from timeout_decorator import timeout
+
 from platforms.openstack import Openstack
 from platforms.vmware import VMware
-from utils import Format
+from utils import (Format, step, Utils)
 
 
 class Platform:
 
-    @staticmethod    
-    def get_platform(conf, platform):
-        if platform.lower() == "openstack":
-            platform = Openstack(conf)
-        elif platform.lower() == "vmware":
-            platform = VMware(conf)
-        elif platform.lower() == "bare-metal":
-            # TODO platform = Bare_metal(conf, utils)
-            print("Todo: bare-metal is not available")
-            sys.exit(0)
-        elif platform.lower() == "libvirt":
-            # TODO platform = Livbirt(conf, utils)
-            print("Todo: libvirt is not available")
-            sys.exit(0)
-        else:
-            raise Exception(Format.alert("Platform Error: {} is not applicable".format(platform)))
-
-        return platform
+    def __init__(self, conf, provisioner):
+        self.conf = conf
+        self.utils = Utils(conf)
+        self.provisioner = provisioner
```

Comment:
Given the current separation of concerns, I think the provisioner should be used by Terraform, not by the Platform class.